.github/workflows/auto-merge.js

```diff
@@ -0,0 +1,107 @@
+module.exports = async ({ github, context }) => {
+  const {
+    repo: { owner, repo },
+  } = context;
+
+  const MERGE_INTERVAL_MS = 5000; // 5 seconds pause after a merge
+
+  async function sleep(ms) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  async function logRateLimit() {
+    const { data: rateLimit } = await github.rest.rateLimit.get();
+    console.log(`Rate limit remaining: ${rateLimit.resources.core.remaining}`);
+    console.log(
+      `Rate limit resets at: ${new Date(rateLimit.resources.core.reset * 1000).toISOString()}`
+    );
+  }
+
+  async function fetchPullRequestDetails(prNumber) {
+    const pullRequest = await github.rest.pulls
+      .get({
+        owner,
+        repo,
+        pull_number: prNumber,
+      })
+      .then((res) => res.data);
+
+    if (pullRequest.mergeable !== null) {
+      return pullRequest;
+    }
+    return null;
+  }
+
+  async function isPRApproved(prNumber) {
+    const { data: reviews } = await github.rest.pulls.listReviews({
+      owner,
+      repo,
+      pull_number: prNumber,
+    });
+    return reviews.some((review) => review.state === "APPROVED");
+  }
+
+  async function areAllChecksPassed(sha) {
+    const { data: checkRuns } = await github.rest.checks.listForRef({
+      owner,
+      repo,
+      ref: sha,
+    });
+    return checkRuns.check_runs.every(
+      (run) => run.conclusion === "success" || run.conclusion === "skipped"
+    );
+  }
+
+  // Get date from a month ago in ISO format
+  const oneMonthAgo = new Date();
+  oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
+  const sinceDate = oneMonthAgo.toISOString();
+
+  // List PRs with the "automerge" label created within the last month
+  const { data: issues } = await github.rest.issues.listForRepo({
+    owner,
+    repo,
+    labels: "automerge",
+    since: sinceDate,
+  });
+
+  // Filter for pull requests from the list of issues
+  const pullRequests = issues.filter((issue) => issue.pull_request);
+
+  for (const pr of pullRequests) {
+    const pullRequest = await fetchPullRequestDetails(pr.number);
+
+    if (!pullRequest || pullRequest.mergeable !== true) {
```

Comment:
```suggestion
    if (!pullRequest?.mergeable) {
```

can we use optional chaining?